src/utils.js

```diff
@@ -12,3 +15,36 @@ export function parseGbToBytes (gbString: string): number | null {
   }
   return parsedGb * (1024 ** 3)
 }
+
+function checkPermissions (allowedPermissions: Array<string>, permissions: Array<PermissionType>): boolean {
+  return permissions.find((role) => (
+    (Selectors.getFilter() || (
+      (role.groupId && Selectors.getUserGroups().includes(role.groupId)) ||
+      (role.userId && role.userId === Selectors.getUserId())
```

Comment:
nit-pick: pull the filter state, userId and groups from the `Selectors` once and hold the data in the function scope